src/pocketmine/plugin/ScriptPluginLoader.php

```diff
@@ -72,24 +97,33 @@ public function getPluginDescription(string $file) : ?PluginDescription{
 			}
 
 			if(preg_match("/^[ \t]+\\*[ \t]+@([a-zA-Z]+)([ \t]+(.*))?$/", $line, $matches) > 0){
-				$key = $matches[1];
+				$key = strtolower($matches[1]);
 				$content = trim($matches[3] ?? "");
 
-				if($key === "notscript"){
-					return null;
-				}
-
 				$data[$key] = $content;
 			}
 
 			if($insideHeader and strpos($line, "*/") !== false){
 				break;
 			}
 		}
-		if($insideHeader){
-			return new PluginDescription($data);
-		}
 
+		return $data;
+	}
+
+	private static function detectMainClass(string $file) : ?string{
+		// It is possible to use token_get_all() to analyze the file syntactically,
+		// but it is usually not necessary, and is too complex for a simple problem.
+		$namespace = "";
+		foreach(file($file) as $line){
+			if(preg_match('/^namespace[ \t]+([A-Za-z0-9_\\]+)/', $line, $match) === 0){
+				$namespace = $match[1] . "\\";
+				continue;
+			}
+			if(preg_match('/^class[ \t]+([A_Za-z0-9_]+)[ \t]+extends\b/', $line, $match) === 0){
```

Comment:
Should change to `^[ \t]+class` to allow indented classes (e.g. under a namespace block)